kiwipy/rmq/communicator.py

```diff
@@ -401,362 +394,27 @@ async def broadcast_send(self, body, sender=None, subject=None, correlation_id=N
-
-    def _run_loop(self, start_future):
-        """Here we are on the comms thread"""
-        _LOGGER.debug('Starting event loop (id %s) on %s', id(self._loop), threading.current_thread())
-
-        self._stop_signal = self._loop.create_future()
-
-        async def do_connect():
-            try:
-                # Connect
-                await self._communicator.connect()
-                start_future.set_result(True)
-            except Exception as exception:  # pylint: disable=broad-except
-                start_future.set_exception(exception)
-            else:
-                # Wait to stop
-                stop_future = await self._stop_signal
-                try:
-                    # Now disconnect
-                    with kiwipy.capture_exceptions(stop_future):
-                        await self._communicator.disconnect()
-                finally:
-                    stop_future.set_result(True)
-
-        self._loop.run_until_complete(do_connect())
-
-        # The loop is finished and the connection has been disconnected
-        self._stop_signal = None  # This should be done before nullifying the communicator_thread
-        self._communicator_thread = None
-
-        _LOGGER.debug('Event loop stopped on %s', threading.current_thread())
-        asyncio.set_event_loop(None)
-
-    def _ensure_open(self):
-        if self.is_closed():
-            raise kiwipy.CommunicatorClosed
-
-
-def connect(connection_params=None,
-            connection_factory=aio_pika.connect_robust,
-            loop=None,
-            message_exchange=defaults.MESSAGE_EXCHANGE,
-            task_exchange=defaults.TASK_EXCHANGE,
-            task_queue=defaults.TASK_QUEUE,
-            task_prefetch_size=defaults.TASK_PREFETCH_SIZE,
-            task_prefetch_count=defaults.TASK_PREFETCH_COUNT,
-            encoder=defaults.ENCODER,
-            decoder=defaults.DECODER,
-            testing_mode=False):
-    # pylint: disable=too-many-arguments
-    return RmqThreadCommunicator.connect(connection_params=connection_params,
-                                         connection_factory=connection_factory,
-                                         loop=loop,
-                                         message_exchange=message_exchange,
-                                         task_exchange=task_exchange,
-                                         task_queue=task_queue,
-                                         task_prefetch_size=task_prefetch_size,
-                                         task_prefetch_count=task_prefetch_count,
-                                         encoder=encoder,
-                                         decoder=decoder,
-                                         testing_mode=testing_mode)
+    async def task_queue(self,
+                         queue_name: str,
+                         prefetch_size=defaults.TASK_PREFETCH_SIZE,
+                         prefetch_count=defaults.TASK_PREFETCH_COUNT) -> tasks.RmqTaskQueue:
```

Comment:
Are you on purpose not allowing to define different encoder/decoder and or the testing mode on a per task-queue basis?

Reply:
Yes.  I consider being in testing more and the encoding scheme to be a per communicator setting (unless the user wants to go lower level and access the `rmq` package directly)